.pre-commit-config.yaml

```diff
@@ -0,0 +1,97 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.4.0
+    hooks:
+      - id: check-added-large-files
+      - id: check-ast
+      - id: check-case-conflict
+      - id: check-merge-conflict
+      - id: debug-statements
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+        args: [--markdown-linebreak-ext=md]
+  - repo: https://github.com/adrienverge/yamllint
+    rev: "v1.26.0"
+    hooks:
+      - id: yamllint
+  - repo: https://github.com/asottile/setup-cfg-fmt
+    rev: v1.17.0
+    hooks:
+      - id: setup-cfg-fmt
+  - repo: https://github.com/psf/black
+    rev: 21.7b0
+    hooks:
+      - id: black
+  - repo: https://github.com/PyCQA/isort
+    rev: "5.9.3"
+    hooks:
+      - id: isort
+  # TODO renable when erros are fixed/ignored
+  # - repo: https://github.com/pycqa/pylint
+  #   rev: "v2.9.6"
+  #   hooks:
+  #     - id: pylint
+  # TODO renable when erros are fixed/ignored
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: "3.9.2"
+    hooks:
+      - id: flake8
+        additional_dependencies:
+          &fd [
+            flake8-annotations-complexity,
+            flake8-bandit,
+            flake8-blind-except,
+            flake8-bugbear,
+            flake8-builtins,
+            flake8-cognitive-complexity,
+            flake8-comprehensions,
+            flake8-docstrings,
+            flake8-eradicate,
+            flake8-executable,
+            flake8-expression-complexity,
+            flake8-if-expr,
+            flake8-implicit-str-concat,
+            flake8-logging-format,
+            flake8-pathlib,
+            flake8-print,
+            flake8-pytest,
+            flake8-pytest-style,
+            # flake8-quotes,  # conflicts with blacks double quote preference
+            flake8-raise,
+            flake8-return,
+            flake8-typing-imports,
+            flake8-variables-names,
+            flake8==3.9.2,
+            pandas-vet,
+            pep8-naming,
+            # wemake-python-styleguide,  # conflicts with black
+            yesqa,
+          ]
+        verbose: true
+        args: &fa [--statistics, --exit-zero]
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.910
+    hooks:
+      - id: mypy
+        additional_dependencies: [types-python-dateutil]
+  - repo: https://github.com/nbQA-dev/nbQA
+    rev: 1.1.0
+    hooks:
+      - id: nbqa-black
+        # Match version of black used for .py and .pynb
+        additional_dependencies: [black==21.7b0]
+      - id: nbqa-isort
+        additional_dependencies: [isort==5.9.3]
+      - id: nbqa-mypy
+        additional_dependencies: [mypy==0.910]
+      # TODO renable when erros are fixed/ignored
+      - id: nbqa-flake8
```

Comment:
Could it be that the notebooks (in docs) are not picked up by these checks?

This check passes on `pre-commit run --all-files` for me, but if I do `pip install nbqa` and then `nbqa flake8 docs/*.ipynb` I get a list of issues.

Reply:
I think the issue is
```
        args: *fa
```
, I'm not sure what it's meant to do here, but if you remove it then `pre-commit run nbqa-flake8 --all-files` will fail too

Reply:
The `flake8` hook is verbose by default while `nbqa-flake8` was not, I forced it to be verbose so now warnings show up.

The `*fa` is a [YAML alias of YAML anchor](https://ktomk.github.io/writing/yaml-anchor-alias-and-merge-key.html) defined in `flake8` hook with `@fa` so I do not have to repeat the list of flake8 plugins.

Reply:
I didn't know you could do that, thanks! This'll be useful in other projects